torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -242,13 +244,33 @@ def do_auto_functionalize(
             "Using `self` or `self_` as an argument in the definition of custom ops may lead to ambiguous parsing. "
             "Please consider using a different name for this argument to avoid potential issues."
         )
-    with ctx.redispatch_to_next():
-        unwrapped_outs = auto_functionalized(
-            op, **unwrapped_kwargs  # type: ignore[arg-type]
-        )
+    
+    # for each mutated arg, identify the items aliased by that args, this is needed for the decomposition to create a 
+    # copy of the aliased items
 
     # List of the name of args that get mutated (according to the schema)
     mutable_args_names = get_mutable_arg_names(op)
+    arg_to_aliased={}
+    for arg in mutable_args_names:
+        arg_to_aliased[arg] = ctx.unwrap_tensors(list(storage_to_tensors[normalized_kwargs[arg]._typed_storage()._cdata]))
+        
+    # take the union of all the aliased items.
+    all_aliased = list(set.union(*map(set, arg_to_aliased.values())))
+    print("all aliased:")
+    print(all_aliased)
+
+    # for each arg, identify the items that
+    with ctx.redispatch_to_next():
+        # The output have the the following structure:
+        # - op output
+        # - mutated 0, alias 0, alias 1, alias 2, alias 3...etc
+        # - mutated kwargs
+        # We only need the first one, so we slice it.
+        print("inputs")
+        print(unwrapped_kwargs)
+        unwrapped_outs = auto_functionalized(
```

Comment:
passing arg_to_aliased and all_aliased result in the following error:
passing them empty is fine
```

Traceback (most recent call last):
  File "/home/lsakka/pytorch/example.py", line 31, in <module>
    f(x)
  File "/home/lsakka/pytorch/torch/_dynamo/eval_frame.py", line 464, in _fn
    return fn(*args, **kwargs)
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 1224, in __call__
    return self._torchdynamo_orig_callable(
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 514, in __call__
    return _compile(
  File "/home/lsakka/.conda/envs/myenv/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 896, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 662, in compile_inner
    return _compile_inner(code, one_graph, hooks, transform)
  File "/home/lsakka/pytorch/torch/_utils_internal.py", line 85, in wrapper_function
    return StrobelightCompileTimeProfiler.profile_compile_time(
  File "/home/lsakka/pytorch/torch/_strobelight/compile_time_profiler.py", line 129, in profile_compile_time
    return func(*args, **kwargs)
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 697, in _compile_inner
    out_code = transform_code_object(code, transform)
  File "/home/lsakka/pytorch/torch/_dynamo/bytecode_transformation.py", line 1322, in transform_code_object
    transformations(instructions, code_options)
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 209, in _fn
    return fn(*args, **kwargs)
  File "/home/lsakka/pytorch/torch/_dynamo/convert_frame.py", line 631, in transform
    tracer.run()
  File "/home/lsakka/pytorch/torch/_dynamo/symbolic_convert.py", line 2722, in run
    super().run()
  File "/home/lsakka/pytorch/torch/_dynamo/symbolic_convert.py", line 957, in run
    while self.step():
  File "/home/lsakka/pytorch/torch/_dynamo/symbolic_convert.py", line 869, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/lsakka/pytorch/torch/_dynamo/symbolic_convert.py", line 2913, in RETURN_VALUE
    self._return(inst)
  File "/home/lsakka/pytorch/torch/_dynamo/symbolic_convert.py", line 2898, in _return
    self.output.compile_subgraph(
  File "/home/lsakka/pytorch/torch/_dynamo/output_graph.py", line 1133, in compile_subgraph
    self.compile_and_call_fx_graph(tx, pass2.graph_output_vars(), root)
  File "/home/lsakka/.conda/envs/myenv/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/lsakka/pytorch/torch/_dynamo/output_graph.py", line 1360, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
  File "/home/lsakka/pytorch/torch/_dynamo/output_graph.py", line 1407, in call_user_compiler
    return self._call_user_compiler(gm)
  File "/home/lsakka/pytorch/torch/_dynamo/output_graph.py", line 1437, in _call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
  File "/home/lsakka/pytorch/torch/_dynamo/repro/after_dynamo.py", line 129, in __call__
    compiled_gm = compiler_fn(gm, example_inputs)
  File "/home/lsakka/pytorch/torch/_dynamo/repro/after_dynamo.py", line 129, in __call__
    compiled_gm = compiler_fn(gm, example_inputs)
  File "/home/lsakka/pytorch/torch/__init__.py", line 2231, in __call__
    return compile_fx(model_, inputs_, config_patches=self.config)
  File "/home/lsakka/.conda/envs/myenv/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 1488, in compile_fx
    return aot_autograd(
  File "/home/lsakka/pytorch/torch/_dynamo/backends/common.py", line 69, in __call__
    cg = aot_module_simplified(gm, example_inputs, **self.kwargs)
  File "/home/lsakka/pytorch/torch/_functorch/aot_autograd.py", line 1019, in aot_module_simplified
    compiled_fn = dispatch_and_compile()
  File "/home/lsakka/pytorch/torch/_functorch/aot_autograd.py", line 1008, in dispatch_and_compile
    compiled_fn, _ = create_aot_dispatcher_function(
  File "/home/lsakka/pytorch/torch/_functorch/aot_autograd.py", line 429, in create_aot_dispatcher_function
    return _create_aot_dispatcher_function(flat_fn, flat_args, aot_config)
  File "/home/lsakka/pytorch/torch/_functorch/aot_autograd.py", line 730, in _create_aot_dispatcher_function
    compiled_fn, fw_metadata = compiler_fn(
  File "/home/lsakka/pytorch/torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py", line 178, in aot_dispatch_base
    compiled_fw = compiler(fw_module, updated_flat_args)
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 1327, in fw_compiler_base
    return _fw_compiler_base(model, example_inputs, is_inference)
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 1394, in _fw_compiler_base
    return inner_compile(
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 457, in compile_fx_inner
    return _compile_fx_inner(*args, **kwargs)
  File "/home/lsakka/pytorch/torch/_dynamo/repro/after_aot.py", line 85, in debug_wrapper
    inner_compiled_fn = compiler_fn(gm, example_inputs)
  File "/home/lsakka/pytorch/torch/_inductor/debug.py", line 313, in inner
    return fn(*args, **kwargs)
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 446, in wrapper
    return fn(*args, **kwargs)
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 644, in _compile_fx_inner
    compiled_graph = FxGraphCache.load(
  File "/home/lsakka/pytorch/torch/_inductor/codecache.py", line 1314, in load
    compiled_graph = compile_fx_fn(
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 554, in codegen_and_compile
    compiled_graph = fx_codegen_and_compile(gm, example_inputs, **fx_kwargs)
  File "/home/lsakka/.conda/envs/myenv/lib/python3.10/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/lsakka/pytorch/torch/_inductor/compile_fx.py", line 836, in fx_codegen_and_compile
    graph.run(*example_inputs)
  File "/home/lsakka/pytorch/torch/_inductor/graph.py", line 768, in run
    return super().run(*args)
  File "/home/lsakka/pytorch/torch/fx/interpreter.py", line 147, in run
    self.env[node] = self.run_node(node)
  File "/home/lsakka/pytorch/torch/_inductor/graph.py", line 1276, in run_node
    result = super().run_node(n)
  File "/home/lsakka/pytorch/torch/fx/interpreter.py", line 204, in run_node
    return getattr(self, n.op)(n.target, args, kwargs)
  File "/home/lsakka/pytorch/torch/_inductor/graph.py", line 1074, in get_attr
    return tensor(value.tolist(), dtype=value.dtype, device=value.device)
  File "/home/lsakka/pytorch/torch/_subclasses/fake_tensor.py", line 881, in tolist
    torch._check_is_size(s)
  File "/home/lsakka/pytorch/torch/__init__.py", line 1571, in _check_is_size
    _check(i >= 0, message)
  File "/home/lsakka/pytorch/torch/__init__.py", line 1557, in _check
    _check_with(RuntimeError, cond, message)
  File "/home/lsakka/pytorch/torch/__init__.py", line 1520, in _check_with
    if expect_true(cond):
  File "/home/lsakka/pytorch/torch/fx/experimental/symbolic_shapes.py", line 986, in expect_true
    return a.node.expect_true(frame.f_code.co_filename, frame.f_lineno)
  File "/home/lsakka/pytorch/torch/fx/experimental/sym_node.py", line 476, in expect_true
    return self.shape_env.defer_runtime_assert(
  File "/home/lsakka/pytorch/torch/fx/experimental/recording.py", line 262, in wrapper
    return retlog(fn(*args, **kwargs))
  File "/home/lsakka/pytorch/torch/fx/experimental/symbolic_shapes.py", line 5451, in defer_runtime_assert
    assert not self.runtime_asserts_frozen, expr
AssertionError: u0 >= 0

```